### PR TITLE
DAOS-6732: Remove line of code that is not reached

### PR DESCRIPTION
**Pull request by marcelarosalesj**
This bug was identified by Coverity

Signed-off-by: Marcela Rosales <marcela.a.rosales.jimenez@intel.com>

### CODE REVIEW

**Review by jolivier23 — CHANGES_REQUESTED** [review]
What is the point of st at all here?   I think we should probably take st as an argument to ts_cmd_run and pass it to the other routines.   Either that or we should get rid of it altogether if it's not needed to pass it to subroutines for a given test.

It looks like there are two calls, one from cmocka and one not...we could pass NULL to the other one but just pass state to the cmocka version.

**Comment by marcelarosalesj** [comment]
> What is the point of st at all here? I think we should probably take st as an argument to ts_cmd_run and pass it to the other routines. Either that or we should get rid of it altogether if it's not needed to pass it to subroutines for a given test.
> 
> It looks like there are two calls, one from cmocka and one not...we could pass NULL to the other one but just pass state to the cmocka version.

Thanks for your feedback @jolivier23. I just sent a patch removing the st fom the other functions because I don't think it is necessary to have it. I reviewed the cmocka documentation and `void **state` is used for CMUnitTest test cases, and after https://github.com/daos-stack/daos/pull/2364, the ts_* functions are not that kind of test cases anymore.

File: src/vos/tests/evt_ctl.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ static struct evt_desc_cbs	ts_evt_desc_nofree_cbs  = {
 };
 
 static void
-ts_open_create(void **state)
+ts_open_create()
```
Comment (jolivier23) [inline src/vos/tests/evt_ctl.c:124]: should be ts_open_create(void)

Same comment for other functions definitions.  This ensures a compiler error if user actually tries to pass a parameter.

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/vos/tests/evt_ctl.c (hunk 1 above)
Comment (marcelarosalesj) [reply src/vos/tests/evt_ctl.c]: okay, I'm sending the patch

**Review by jolivier23 — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Pull request merged**